### DISCUSSION TITLE
Renaming suggestions

### DISCUSSION THREAD

**Discussion by AprilSylph** [AprilSylph/XKit-Rewritten#150] [Lounge]
Due to confusion stemming from "*New* XKit" now being the old and broken one, plus the fact that this addon is not comprised of userscripts, I wish to divorce from the "XKit" moniker entirely if possible and rename this addon to something entirely different, but I'm having trouble with thinking of anything good.

Some rules which must be abided by:
1. Must not include the name "Tumblr" as per [Tumblr's trademark guidelines](https://tumblr.zendesk.com/hc/en-us/articles/231697227-Trademark-Guidelines) for safety
2. Must not be appended with "for Tumblr"; needs to stand out by itself
3. Must not otherwise be able to be confused for another product or service
4. Obviously, must not include the name "XKit"

My thought process so far is that "Tumblr Enhancement Suite" would be good if it didn't break rule no. 1 and wasn't a complete ripoff of "Reddit Enhancement Suite" (thus breaking rule no. 3) and that "T-Suite" would be a good compromise if it couldn't be initialised to "TS" (also breaking rule no. 3).

Serious replies only, please. I've heard every "New New XKit" joke already.

**Comment by trisckit** [comment]
ZKit could be an easy one, it’s just one letter after X and shows it a successor to the old “New XKit” (though I don’t actually know the original etymology of XKit so that might not even make sense)

**Reply by AprilSylph** [reply]
XKit is basically short for XenixKit, Xenix being @atesh's pseudonym. ZKit is actually what it's named for me locally, but I'm not sure I want to stick so close to "XKit"

**Comment by nightpool** [comment]
(doesn't "Tumblr Enhancement Suite" violate rule 1?)

**Reply by AprilSylph** [reply]
i guess it does. fuck

**Comment by nightpool** [comment]
I know this already violates one of your rules, but my actual serious suggestion is that we should just call it XKit 8, because it's the sequel to XKit and it's the ground-up rewrite that XKit 8 was always intended to be. The only feature missing from Rewritten that I wanted to include in XKit 8 is shareable, user-submitted scripts ala greasyfork or the Firefox extension gallery.

**Reply by nightpool** [reply]
Anyway, I understand if you want to go in a different direction and distance yourself from XKit, and if you do I would encourage you to do so, but I just want to say that the option is there if you want to take it.

**Reply by AprilSylph** [reply]
You've said before that it not including userscript capabilities should disqualify it from being called XKit at all, though. Have you changed your mind, or are you still hoping for userscripts? Because I'm not interested in doing userscripts. I am very interested in distributing the utilities the contentscripts rely on separately, but I don't know if it would be valuable to userscript writers at all. XKit 7 is the only userscript framework I've ever used successfully.

**Reply by nightpool** [reply]
hmm I did say that, didn't I. 

on one hand I don't know if I *disagree* with the sentiment I previously had, but I do want to be practical about things, because like. A cool, working XKit that doesn't include userscripts is way better then a shitty, broken one that does. And nobody really cares about userscripts except for my weird ideological attachment to them anyway.

When you say that you don't ever want to do userscripts in rewritten, is that *primarily* about the multiple-core-extensions-that-people-install thing that New XKit has going on (which I agree is bad), or is it more about actual user scripts that aren't in GitHub, or both?

**Reply by nightpool** [reply]
how would you feel about the kind of system I proposed in my first comment?

**Reply by AprilSylph** [reply]
It's primarily that there is no approved cross-browser userscript implementation. Firefox WebExtensions have their own system which isn't supported in Chromium, and I don't know what Chromium userscript addons do but if it's eval that's going to get axed with manifest v3. I suppose it would be easy to implement a userscript system if it just dumped everything into the page context, but that would involve putting all the utilities in the page context as well... not sure how easy that would be.

**Reply by nightpool** [reply]
If that's your main objection, I'm confident of our ability to solve that problem or cross that bridge when we come to it. Whatever manifest v3 ends up doing, they're not going to throw out Tampermonkey.

**Reply by RepairedCheese** [reply]
Sounds to me like it's still XKit, even if it's being written from scratch. And that's mostly because this isn't just a new, fresh project. It's targeting XKit features. Back in the days of Missing e, I do recall that it and XKit mostly avoided covering the same features, until Missing e died.

Userscript or no, this is going to be the sequel to New Xkit.  And perhaps the userscript can come later, besides.

**Comment by mechagotch** [comment]
~~(writing a serious suggestion now because I realise a joke was in bad taste considering you asked specifically for serious replies)~~
Synonyms for 'tumbler' could be useful for a full rename. Flask Extensions/Flask Extension Suite sounds nice (though, Flask is the name of a python framework which could cause minor confusion, though probably less with the general Tumblr userbase)

**Reply by AprilSylph** [reply]
I tried going down this route before, but wasn't sure if enough people would connect Tumblr=Tumbler=Flask (i remember only learning the word "tumbler" from the Valve merch store, and haven't seen it since). I suppose I should give it a lot more thought though, considering how "XKit" has no relation to Tumblr *at all*.

**Comment by hobinjk** [comment]
Better Dash feels like it could almost work (sure, Tumblr is much more than the dash but the most visible modifications are on the dash). Also speaking for myself I'd be willing to pursue rebranding New XKit as Old XKit or equivalent if it helps.

**Reply by AprilSylph** [reply]
I think it'd be helpful to rename New XKit to XKit 7. If that can be done maybe I will just end up going with "XKit 8", though I probably wouldn't rename this repository, just change the addon title.

**Comment by marcustyphoon** [comment]
Here are my full thoughts so far:

- I personally don't mind "kit" as a term used to describe "a collection of tools," whether or not users can distribute their own.

- Barring getting special permission from Tumblr, I understand why the "no 'Tumblr'" rule exists, from their perspective. I think most ways around this that attempt to make it clear what the extension is for, like using the letter T/puns/etc, will draw attention to the rule and feel really forced. Missing E really was the perfect name.

- The only ways I can think of around this are using something incredibly distinctive to tumblr ("reblog"? "dashboard" is marginal but a possibility) or keeping "XKit" around since people know what that is already.

- Ideally it should be clear that this is the next iteration of the extension, not a fork or old version or competing extension or etc. Using a number does this well, imo, as even without 7 being visibly numbered, it inherently begs the question, "am I currently using XKit 7?" I haven't thought of a non-number way to hint at this yet. Honorable mention to parodying Apple with "XKit (2021)." Even if there were a great superlative to supercede "New," I don't know how not to run into this.

So I'm trying to poke a hole in or think of a way around any of these. Not having yet done so, "XKit 8" is the option I see the least wrong with. "ReblogKit" kinda works for most of the rules, but I really don't like it for some reason :D It's just less clear then "XKit Rewritten" for no benefit.

(edit from much later for those looking back on this: this edit breaks the "serious suggestions only" rule, but it should tooootally just still be `x8` - it breaks 0 rules! Except, y'know, the one where you have to be able to tell what on earth it is.)

**Comment by cyle** [comment]
> Must not be appended with "for Tumblr"; needs to stand out by itself

not sure i understand this restriction, since the thing only works..... _for tumblr_. it's not like you could use it for reddit or twitter, right?

**Reply by marcustyphoon** [reply]
I assume that's more a "can't be something long enough to be a standalone title, but confusing enough to need to be clarified," i.e. "AprilsCoolNewExtension for Tumblr."

"Tweaks for Tumblr," like you suggested, is actually a decent idea, if very generic. Might not be obvious in casual conversation that it's a title and not a description.

**Comment by paints-with-bold-colours** [comment]
~~"Improvements for Hellsite (Affectionate)"~~

How about DashKit? Dash evokes tumblr for most people, I think. Plus as marcus said, "kit" is a nice word for a set of tools, and keeping it will remind users that this has similar functions as xkit, while being easily distinguishable thanks to that first syllable.

**Reply by nightpool** [reply]
Blue Hellsite Enhancement Suite is kind of funny to imagine

**Reply by invalidCards** [reply]
I like this suggestion, but I am afraid that people will see the word "kit" in the context of a Tumblr-enhancing extension and immediately associate it with (New) XKit, which probably has its own pros and cons. I'd personally rather go with something like DashPack - it has generally the same meaning without the association.

**Reply by paints-with-bold-colours** [reply]
I was thinking of an association with xkit as a positive because it'd help users know what types of features to expect, but I suppose there could be cons to it as well.

**Reply by Nvara-of-Mortain-s-Own** [reply]
What about tunglr ZKit? Is that too close to tumblr to be permissible?
~~hellsite enhancement suite has my vote~~

**Comment by cyle** [comment]
some ideas:

- Tweaks for Tumblr (because i think "for tumblr" is valid)
- Missing E (reclaim it!)
- Missing Period (since we dropped the period from the logo)
- rlbmut (get it?)
- 'blr toolkit
- BlrTools
- Blur (playing off 'blr)
- Tjamblur (what i call tumblr internally)

**Reply by nightpool** [reply]
"Missing Period" I think would be ..... unwise.

**Reply by cyle** [reply]
yes. it would be.

**Comment by paints-with-bold-colours** [comment]
Couple other things cobbled together from other ideas here
Dash Tools
Dash Enhancements
Dash Enhancer
Dash Tweaks

**Comment by FishstickPaladin** [comment]
What about “Tweakd” (like how Tumblr leaves out the ‘e’ for “tumbler” but with “tweaked”)?

**Reply by nightpool** [reply]
I like this. this is the best suggestion yet

**Reply by invalidCards** [reply]
+1 from me too 😄

**Reply by arcanetrivia** [reply]
That's not bad.

I would avoid putting "reblog" in the name of the whole package because then it tends to make it sound like that's the only function it addresses.
